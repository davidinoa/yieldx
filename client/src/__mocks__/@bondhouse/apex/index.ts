import { AccountStatus } from "@bondhouse/apex";
import { MOCK_APEX_ACCOUNT } from "./__fixtures__";
// eslint-disable-next-line @typescript-eslint/no-empty-function,no-shadow
function ApexAccountsApi() {}

ApexAccountsApi.prototype.getApexAccountByInvestorId = jest.fn(() =>
  Promise.resolve({
    data: {
      apexAccount: MOCK_APEX_ACCOUNT
    }
  })
);

// eslint-disable-next-line @typescript-eslint/no-empty-function,no-shadow
function StatementsApi() {}

StatementsApi.prototype.getAccountStatements = jest.fn(() =>
  Promise.resolve([
    {
      apexAccountId: "28b5b427-053b-4ad8-9acb-05a5e7bbe614",
      statmentDate: "12/31/2014",
      documentType: "Account Statement",
      statementURL:
        "https://UATEDocuments.apexclearing.com/RedirectConfirmURL.aspx?ds=1DC69CF82426F431D4535F6549270DAC58EEB05D61A8209E49CAAE3BC76EE8B866CEDC5A04A68284717045A7352477CC49D5E186BA7CA707C0EF8A692C654BED5732532662D182181AAF4E3C7F5B219956BF0F1B6F57C588C0465DF32D3A5EB58EFB4CFF188B33275E8BFB631B97C80DFEB0838AECFC00DF537E96EB28868EFF10660E83DD1A6C5808E6DED036C15E25336F448E9F69DF30118E6960DD1F98D4B8E4714BCAE3D1435A79505BC3FB730B09FEA86B8E3DEB1F8DDAF65E746FBE833308839C1FA0381057F4E71EFC7E14AF5DBB327B2CA848EC4CC2D036AF66C648E78AD70688C550DEB37F942CC82AEEE4F6E1BA95C082DB8D326BC86D5B5DC113060E7449A97FD2F019A66050CA0AF83B0FA50FFCFE9F5E1664C92DA902988BDF&pid=TkVXU1RNVHx2NzEyNi01Mj9yLTUyOTMtNzc1xS1PREEyLTI4MzBGQUFZLTM5NzAxLTU0MzIxNC00MjE4NzItMjE5MjAyLTg1LTc5LTI2NDMtODktMC1eATAxNAE1UEkBMzIyNzIBTUFSSyBTTEFWT05JQQExN1EwNgFXRUEBAgECAQIBAgExNDQBMTcxMDgBMTMxODQBAgECAQIBA0101816"
    }
  ])
);

export { ApexAccountsApi, AccountStatus, StatementsApi };
