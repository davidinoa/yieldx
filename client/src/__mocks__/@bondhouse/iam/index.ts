import {
  CreateInvestorMappingRequest,
  UserType,
  UserStatus
} from "@bondhouse/iam";

// eslint-disable-next-line @typescript-eslint/no-empty-function
export function UsersApi() {}

UsersApi.prototype.getCurrentUser = jest.fn(() =>
  Promise.resolve({
    data: {
      user: {
        id: "auth0|5ec6119474dfa00cbf1d53cb",
        first: "Institution",
        last: "New User",
        userType: "ADMIN",
        email: "test4@bondhouse.com",
        terms: true,
        institutionId: "bf8a5597-9569-4510-babd-a6d9655e9f0d",
        createdAt: "2020-05-21T05:28:52.982380Z",
        updatedAt: "2020-05-21T05:28:52.982380Z",
        status: "ACTIVE",
        policies: []
      }
    }
  })
);

// eslint-disable-next-line @typescript-eslint/no-empty-function
export function PasswordResetTicketsApi() {
  return {};
}

PasswordResetTicketsApi.prototype.createPasswordResetTicket = jest.fn(() =>
  Promise.resolve({
    data: {
      success: true
    }
  })
);

// eslint-disable-next-line @typescript-eslint/no-empty-function
export function InstitutionInventoryMappingApi() {}

InstitutionInventoryMappingApi.prototype.getInstitutionInventoryMapping = jest.fn(
  () =>
    Promise.resolve({
      data: {
        inventoryMapping: {
          institutionId: "b17c644a-30e8-47fb-a26b-f6743786c3ba",
          inpaas: "inpaas",
          bestFit: "bestfit",
          assetExplorer: "assetexplorer"
        }
      }
    })
);
InstitutionInventoryMappingApi.prototype.createOrUpdateInstitutionInventoryMapping = jest.fn();

// eslint-disable-next-line @typescript-eslint/no-empty-function
export function InstitutionInvestorMappingApi() {}

InstitutionInvestorMappingApi.prototype.createInstitutionInvestorMapping = jest.fn(
  (
    institutionId: string,
    createInvestorMappingRequest?: CreateInvestorMappingRequest
  ) =>
    Promise.resolve({
      data: {
        investorMapping: {
          institutionId,
          investorId: createInvestorMappingRequest?.investorId || ""
        }
      }
    })
);

export { UserType, UserStatus };
