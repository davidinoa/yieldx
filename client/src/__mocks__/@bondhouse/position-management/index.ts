import {
  CreateFundTransferResponse,
  CreateOrReplaceExternalPortfolioResponse,
  LedgerEntryStatusEnum,
  PortfolioStatus,
  AppType
} from "@bondhouse/position-management";

import samplePortfolio from "util/samplePortfolio";
import usdCash from "util/usdCash";

export function PortfoliosApi() {
  return {
    async internalPortfolioFundTransfer() {
      return {
        accepted: true,
        ledgerEntry: {
          id: "",
          tradeId: "",
          portfolioId: "",
          assetId: "",
          quantity: 1,
          tradePrice: 1,
          counterPartyId: "",
          relatedPositions: [],
          status: LedgerEntryStatusEnum.SETTLED
        }
      } as CreateFundTransferResponse;
    },

    updatePortfolio: jest.fn(async () => ({
      id: "mockId",
      createdAt: new Date().toISOString(),
      updateAt: new Date().toISOString(),
      status: PortfolioStatus.TERMINATED,
      currency: "USD",
      name: "Mock Portfolio",
      positions: [usdCash]
    })),

    getPortfolio: jest.fn(async () => {
      return {
        data: {
          portfolio: {
            ...samplePortfolio,
            positions: []
          }
        }
      };
    }),

    createPortfolioForInvestor: jest.fn(async () => {
      const portfolio = {
        ...samplePortfolio,
        positions: []
      };
      delete portfolio.metadata;
      return { data: { portfolio } };
    })
  };
}

export function TradesApi() {
  return {
    getTrades: jest.fn(async () => ({
      data: {
        ledgerEntries: []
      }
    }))
  };
}

export function ExternalPortfolioApi() {
  return {
    createExternalPortfolio: jest.fn(async () => {
      return {
        created: true,
        updated: false
      } as CreateOrReplaceExternalPortfolioResponse;
    }),
    getExternalPortfolio: jest.fn(async () => {
      return {
        data: {
          portfolio: { positions: [] }
        }
      };
    })
  };
}

export { AppType, PortfolioStatus, LedgerEntryStatusEnum };
