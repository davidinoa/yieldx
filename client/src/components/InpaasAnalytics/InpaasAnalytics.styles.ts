import styled from "styled-components";
import { Row } from "react-bootstrap";

export const AnalyticsRow = styled(Row)`
  min-height: 296px;
`;
