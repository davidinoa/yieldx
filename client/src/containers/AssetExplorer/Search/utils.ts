interface InGridFilter {
  filter: string | number;
  filterTo: number;
  filterType: string;
  type: string;
  dateFrom: string;
  dateTo: string;
  values: string[];
}

export interface AgGridFilters {
  [key: string]: any;
}

const getFilter = (filter: InGridFilter) => {
  /**
   *  The filter object from Ag Grid API. In the case of filtering `name` column.
   *  { filter: "Asset Name", filterType: "text", type: "contains"}
   */
  let searchFilter = {};
  const {
    filter: filterContent,
    filterType,
    type,
    filterTo,
    values,
    dateFrom,
    dateTo
  } = filter;
  if (filterType === "text") {
    switch (type) {
      case "contains":
        searchFilter = { containsText: filterContent };
        break;
      case "notContains":
        searchFilter = { doesNotContainText: filterContent };
        break;
      case "equals":
        searchFilter = { equalsText: filterContent };
        break;
      case "notEqual":
        searchFilter = { doesNotEqualText: filterContent };
        break;
      case "startsWith":
        searchFilter = { startsWithText: filterContent };
        break;
      case "endsWith":
        searchFilter = { endsWith: filterContent };
        break;
      default:
        searchFilter = { containsText: filterContent };
        break;
    }
  } else if (filterType === "number") {
    switch (type) {
      case "equals":
        searchFilter = { equalsNumber: filterContent };
        break;
      case "notEqual":
        searchFilter = { doesNotEqualNumber: filterContent };
        break;
      case "lessThan":
        searchFilter = { isLessThanNumber: filterContent };
        break;
      case "lessThanOrEqual":
        searchFilter = { isLessThanOrEqualToNumber: filterContent };
        break;
      case "greaterThan":
        searchFilter = { isGreaterThanNumber: filterContent };
        break;
      case "greaterThanOrEqual":
        searchFilter = { isGreaterThanOrEqualToNumber: filterContent };
        break;
      case "inRange":
        searchFilter = {
          isGreaterThanOrEqualToNumber: filterContent,
          isLessThanOrEqualToNumber: filterTo
        };
        break;
      default:
        searchFilter = { equalsNumber: filterContent };
        break;
    }
  } else if (filterType === "date") {
    const convertedDateFrom = new Date(dateFrom);
    const convertedDateTo = new Date(dateTo);
    switch (type) {
      case "equals":
        searchFilter = { equalsDate: convertedDateFrom };
        break;
      case "greaterThan":
        searchFilter = { isGreaterThanDate: convertedDateFrom };
        break;
      case "lessThan":
        searchFilter = { isLessThanDate: convertedDateFrom };
        break;
      case "notEqual":
        searchFilter = { doesNotEqualDate: convertedDateFrom };
        break;
      case "inRange":
        searchFilter = {
          isGreaterThanOrEqualToDate: convertedDateFrom,
          isLessThanOrEqualToDate: convertedDateTo
        };
        break;
      default:
        searchFilter = { equalsDate: convertedDateFrom };
        break;
    }
  } else if (filterType === "set" && values) {
    // We may need to add `set` type Ag grid column filter params like cusip or isin
    // in the search request in the future.
    // searchFilter = { inText: values };
  }
  return searchFilter;
};

export const convertInGridFilters = (filters: AgGridFilters) => {
  const convertedFilters: { [index: string]: any } = {};
  Object.keys(filters).forEach(key => {
    const inGridFilter = filters[key];
    if (key === "minimumQuantity") key = "bond.minimumDenomination";
    convertedFilters[key] = getFilter(inGridFilter);
  });

  return convertedFilters;
};
