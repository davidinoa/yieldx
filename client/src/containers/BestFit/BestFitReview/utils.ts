import { getRatingValue } from "../../../util/ratings";
import YieldXPortfolio, {
  YieldXPosition
} from "../../../models/YieldXPortfolio";

export function getBondRatingsData(
  portfolio: YieldXPortfolio
): { name: string; value: number }[] {
  if (portfolio.getBondSleeve) {
    const ratingMarketValues: { [key: string]: number } = {};
    const bondPortfolio = portfolio.getBondSleeve();
    bondPortfolio.positions.forEach((position: YieldXPosition) => {
      const { asset, weight } = position;
      const rating = asset.rating ?? "NR";
      ratingMarketValues[rating] = (ratingMarketValues[rating] || 0) + weight;
    });
    return Object.entries(ratingMarketValues)
      .map(([rating, weight]) => {
        return {
          name: rating,
          value: weight
        };
      })
      .sort((a, b) => {
        return getRatingValue(a.name) - getRatingValue(b.name);
      });
  }
  return [];
}
