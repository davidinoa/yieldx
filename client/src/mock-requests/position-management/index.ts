import { createGetPortfolioHandler } from "./createGetPortfolioHandler";

export default {
  defaultGetPortfolio: createGetPortfolioHandler
};
