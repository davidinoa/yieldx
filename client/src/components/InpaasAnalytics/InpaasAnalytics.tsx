import React from "react";
import { Col } from "react-bootstrap";
import { ToolTipContent } from "assets/content";
import { Comparison } from "@bondhouse/income-portfolios";
import YieldContribution from "containers/Portfolios/PortfolioDisplay/Charts/YieldContribution/YieldContribution";
import PortfolioSection from "containers/Portfolios/PortfolioDisplay/PortfolioSection/PortfolioSection";
import { YieldXPosition } from "models/YieldXPortfolio";
import { calculateYieldContributions } from "util/calculateYieldContributions";
import SectorAllocation from "containers/Portfolios/PortfolioDisplay/Charts/SectorAllocation/SectorAllocation";
import RiskComparison from "components/InpaasAnalytics/RiskComparison/RiskComparison";
import ExpenseRatioComparison from "components/InpaasAnalytics/ExpenseRatioComparison/ExpenseRatioComparison";
import { AnalyticsRow } from "./InpaasAnalytics.styles";

interface Props {
  positions: Array<YieldXPosition>;
  totalMarketValue: number;
  cashMarketValue: number;
  annualYield?: number;
  riskComparison?: Comparison;
  expenseComparison?: Comparison;
}

function InpaasAnalytics({
  positions,
  totalMarketValue,
  cashMarketValue,
  annualYield,
  riskComparison,
  expenseComparison
}: Props) {
  const contribution = !annualYield
    ? undefined
    : calculateYieldContributions(positions, annualYield);
  return (
    <PortfolioSection title="Analytics">
      <AnalyticsRow className="mb-2">
        <RiskComparison comparison={riskComparison} />
        <ExpenseRatioComparison
          portfolioMarketValue={totalMarketValue}
          comparison={expenseComparison}
        />
      </AnalyticsRow>
      <AnalyticsRow>
        <Col xs={12}>
          <SectorAllocation
            type="funds"
            positions={positions}
            totalMarketValue={totalMarketValue}
            cashMarketValue={cashMarketValue}
            tooltipContent={ToolTipContent.SectorAllocation}
          />
        </Col>
        <Col xs={12}>
          <YieldContribution
            contribution={contribution}
            tooltipContent={ToolTipContent.YieldContribution.Funds}
          />
        </Col>
      </AnalyticsRow>
    </PortfolioSection>
  );
}

export default InpaasAnalytics;
