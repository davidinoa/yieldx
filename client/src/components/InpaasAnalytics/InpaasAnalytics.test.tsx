import React from "react";
import { AppType } from "@bondhouse/position-management";
import { buildMockPortfolio, render } from "test/utils";
import InpaasAnalytics from "components/InpaasAnalytics/InpaasAnalytics";

const portfolio = buildMockPortfolio({ appType: AppType.INPAAS });

describe("InpaasAnalytics", () => {
  it("Should render", () => {
    const { container } = render(
      <InpaasAnalytics
        positions={portfolio.positions}
        totalMarketValue={portfolio.totalMarketValue}
        cashMarketValue={portfolio.cashMarketValue}
        annualYield={5.102141837493456}
        expenseComparison={{
          comparisonType: "Expense Ratio",
          inpaas: 0.46454842661307716,
          average: 1.04,
          difference: 0.5754515733869229,
          percentageDifference: 55.33188205643489
        }}
        riskComparison={{
          comparisonType: "Risk",
          inpaas: 6.744287242092945,
          average: 12.9,
          difference: 6.155712757907056,
          percentageDifference: 47.7187035496671
        }}
      />
    );
    expect(container).toBeInTheDocument();
  });

  it("Should match snapshot", () => {
    const { asFragment } = render(
      <InpaasAnalytics
        positions={portfolio.positions}
        totalMarketValue={portfolio.totalMarketValue}
        annualYield={5.102141837493456}
        expenseComparison={{
          comparisonType: "Expense Ratio",
          inpaas: 0.46454842661307716,
          average: 1.04,
          difference: 0.5754515733869229,
          percentageDifference: 55.33188205643489
        }}
        riskComparison={{
          comparisonType: "Risk",
          inpaas: 6.744287242092945,
          average: 12.9,
          difference: 6.155712757907056,
          percentageDifference: 47.7187035496671
        }}
        cashMarketValue={portfolio.cashMarketValue}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
