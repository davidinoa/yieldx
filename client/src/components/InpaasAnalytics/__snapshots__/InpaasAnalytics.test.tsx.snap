// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InpaasAnalytics Should match snapshot 1`] = `
<DocumentFragment>
  .c11 {
  width: 100%;
  background: var(--dark);
  border: 1px solid var(--border-new);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: stretch;
  -webkit-justify-content: stretch;
  -ms-flex-pack: stretch;
  justify-content: stretch;
  border-radius: 0.5rem;
  padding: 1rem 1.15rem;
  min-height: 6rem;
}

.c11 > * {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c13 {
  font-weight: bold;
  line-height: normal;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  text-transform: uppercase;
  color: var(--lightest);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  font-size: 0.65rem;
  padding: 0.25rem 0;
  margin-bottom: 0.25rem;
  max-height: 2rem;
}

.c13 > svg {
  margin: -0.5rem 0 0 0.35rem;
  vertical-align: middle;
  width: 0.85rem;
  height: 0.85rem;
}

.c6 > svg {
  margin: -0.5rem 0 0 0.35rem;
  vertical-align: middle;
  width: 0.85rem;
  height: 0.85rem;
}

.c18 {
  height: 100%;
}

.c18 li > span {
  display: inline-block;
  margin: 0 15px;
}

.c15 {
  position: relative;
}

.c17 {
  width: 100%;
  list-style: none;
  cursor: pointer;
  margin: auto;
}

.c17 > li.hidden {
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
}

.c16 {
  height: 250px;
}

.c0 {
  margin-bottom: 3rem;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c0 .analytics-card-container {
  min-height: 20rem;
  margin-bottom: 0.85rem;
}

.c0 .analytics-card-container > div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
}

.c1 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.c1 h2 {
  margin: 0;
  font-size: 1.15rem;
}

.c1 svg {
  margin: -0.5rem 0 0 0.35rem;
  vertical-align: middle;
  width: 0.85rem;
  height: 0.85rem;
}

.c2 span {
  border: solid 1px var(--border-new);
  color: var(--primary);
  border-radius: 0.25rem;
  font-weight: bold;
  padding: 0.25rem 0.75rem;
  font-size: 0.75rem;
}

.c2 span b {
  font-weight: bold;
  color: var(--lightest);
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  font-size: 0.5rem;
}

.c12 {
  height: 100%;
}

.c12 li > span {
  display: inline-block;
  margin: 0 15px;
}

.c14 {
  margin: 0;
  padding: 0;
}

.c4 {
  width: 100%;
  height: 100%;
  background: var(--dark);
  border: 1px solid var(--border-new);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: stretch;
  -webkit-justify-content: stretch;
  -ms-flex-pack: stretch;
  justify-content: stretch;
  border-radius: 0.5rem;
  padding: 1rem 1.15rem;
  min-height: 6rem;
}

.c4 > * {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c5 {
  font-weight: bold;
  line-height: normal;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  text-transform: uppercase;
  color: var(--lightest);
  position: relative;
  font-size: 0.65rem;
  padding: 0.25rem 0;
  margin-bottom: 0.25rem;
  max-height: 2rem;
}

.c5 > svg {
  margin: -0.5rem 0 0 0.35rem;
  vertical-align: middle;
  width: 0.85rem;
  height: 0.85rem;
}

.c8 {
  line-height: normal;
  font-size: 1.5rem;
}

.c8 sup {
  font-size: 1rem;
  padding: 0.25rem;
}

.c8 span {
  color: var(--subtext-3);
  font-size: 12px;
  line-height: 15px;
}

.c8 .allocation-separator {
  height: 23px;
  width: 2px;
  background: var(--border-new);
}

.c10 {
  position: relative;
  background: var(--page);
  border-radius: 18px;
  height: 24px;
}

.c10 .toggle-button {
  background: none;
  z-index: 1;
  border: none;
  padding: 7px;
  height: 18px;
  font-weight: bold;
  font-size: 0.9rem;
  line-height: 12px;
  -webkit-letter-spacing: 0.1em;
  -moz-letter-spacing: 0.1em;
  -ms-letter-spacing: 0.1em;
  letter-spacing: 0.1em;
  text-transform: uppercase;
}

.c10::before {
  content: "";
  background: var(--light);
  width: 50%;
  position: absolute;
  border-radius: 12px;
  -webkit-transition: all 200ms ease-out;
  transition: all 200ms ease-out;
  top: 3px;
  height: 18px;
  left: 2px;
}

.c10.toggled::before {
  -webkit-transition: all 200ms ease-out;
  transition: all 200ms ease-out;
  left: 47%;
}

.c7 .comparison-label {
  font-family: var(--font-secondary);
  font-weight: bold;
  font-size: 11px;
  line-height: 13px;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  text-transform: uppercase;
  color: var(--subtext-3);
}

.c7 .horizontal-separator {
  height: 2px;
  width: 80%;
  background: var(--border-new);
}

.c7 .vertical-separator {
  height: 80%;
  width: 2px;
  background: var(--border-new);
}

.c7 .comparison-value-container {
  margin: auto auto 0 auto;
}

.c9 .comparison-label {
  font-family: var(--font-secondary);
  font-weight: bold;
  font-size: 11px;
  line-height: 13px;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  text-transform: uppercase;
  color: var(--subtext-3);
}

.c9 .comparison-toggle {
  top: 0;
  right: 50%;
  -webkit-transform: translateX(50%);
  -ms-transform: translateX(50%);
  transform: translateX(50%);
}

.c9 .toggle-button {
  font-size: 10px;
  width: 43px;
}

.c9 .toggle-button:first-of-type {
  margin-left: 5px;
}

.c9 .toggle-button:last-of-type {
  margin-right: 5px;
}

.c9 .vertical-separator {
  height: 80%;
  width: 2px;
  background: var(--border-new);
}

.c3 {
  min-height: 296px;
}

@media (min-width:1600px) {
  .c11 {
    border-radius: 0.625rem;
    padding: 1.65rem;
  }
}

@media (min-width:1600px) {
  .c13 {
    font-size: 0.75rem;
    margin-bottom: 0.5rem;
  }
}

@media (min-width:1600px) {
  .c13 > svg {
    width: 1rem;
    height: 1rem;
  }
}

@media (min-width:1600px) {
  .c6 > svg {
    width: 1rem;
    height: 1rem;
  }
}

@media (min-width:1600px) {
  .c1 h2 {
    font-size: 1.5rem;
  }
}

@media (min-width:1600px) {
  .c1 svg {
    width: 1rem;
    height: 1rem;
  }
}

@media (min-width:1600px) {
  .c2 span {
    font-size: 0.9rem;
  }
}

@media (min-width:1600px) {
  .c2 span b {
    font-size: 0.725rem;
  }
}

@media (min-width:1600px) {
  .c4 {
    border-radius: 0.625rem;
    padding: 1rem 1.65rem 1.65rem 1.65rem;
  }
}

@media (min-width:1600px) {
  .c5 {
    font-size: 0.75rem;
    margin-bottom: 0.5rem;
  }
}

@media (min-width:1600px) {
  .c5 > svg {
    width: 1rem;
    height: 1rem;
  }
}

@media (min-width:1600px) {
  .c8 {
    font-size: 1.95rem;
  }
}

@media (min-width:1600px) {
  .c8 sup {
    font-size: 1.15rem;
  }
}

@media (min-width:1600px) {
  .c10 {
    border-radius: 24px;
    height: 32px;
  }
}

@media (min-width:1600px) {
  .c10 .toggle-button {
    height: 24px;
  }
}

@media (min-width:1600px) {
  .c10::before {
    top: 4px;
    height: 24px;
  }
}

<div
    class="container-fluid"
  >
    <section
      class="c0 container-fluid"
    >
      <div
        class="mb-2 row"
      >
        <div
          class="c1 col-16"
        >
          <h2>
            AnalyticsÂ 
          </h2>
        </div>
        <div
          class="c2 col-8 d-flex align-items-center justify-content-end"
        />
      </div>
      <div
        class="c3 mb-2 row"
      >
        <div
          class="mb-1 col-12"
          data-testid="Portfolio-Metric-Card"
        >
          <div
            class="c4"
          >
            <div
              class="c5 d-flex align-items-center"
            >
              Risk comparison
              <i
                class="c6"
              >
                <svg>
                  info.svg
                </svg>
              </i>
            </div>
            <div
              class="c7 row"
            >
              <div
                class="col-7 d-flex flex-column justify-content-between"
              >
                <div
                  class="comparison-value-container"
                >
                  <div
                    class="c8"
                  >
                    47.72
                    <sup>
                      %
                    </sup>
                  </div>
                  <span
                    class="comparison-label"
                  >
                    LESS RISK
                  </span>
                </div>
                <div
                  class="horizontal-separator m-auto"
                />
                <div
                  class="comparison-value-container mt-0 mb-auto"
                >
                  <div
                    class="c8"
                  >
                    616
                    <sup>
                      bps
                    </sup>
                  </div>
                  <span
                    class="comparison-label"
                  >
                    REDUCTION
                  </span>
                </div>
              </div>
              <div
                class="col-1 d-flex justify-content-center align-content-center"
              >
                <div
                  class="vertical-separator m-auto"
                />
              </div>
              <div
                class="col-16"
              >
                <div
                  style="width: 100%; height: 100%;"
                >
                  <div
                    style="width: 0px; height: 0px;"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="mb-1 col-12"
          data-testid="Portfolio-Metric-Card"
        >
          <div
            class="c4"
          >
            <div
              class="c5 d-flex align-items-center"
            >
              Expense ratio comparison
              <i
                class="c6"
              >
                <svg>
                  info.svg
                </svg>
              </i>
            </div>
            <div
              class="c9 row"
            >
              <div
                class="col-7 d-flex flex-column justify-content-center"
              >
                <div
                  class="c10 d-flex align-items-center mt-4 position-absolute comparison-toggle"
                >
                  <button
                    class="d-flex justify-content-center align-items-center toggle-button"
                    data-testid="ToggleValue-Percent"
                    type="button"
                  >
                    CASH
                  </button>
                  <button
                    class="d-flex justify-content-center align-items-center toggle-button"
                    data-testid="ToggleValue-Dollar"
                    type="button"
                  >
                    BPS
                  </button>
                </div>
                <div
                  class="c8"
                >
                  $0
                  <sup>
                    .00
                  </sup>
                </div>
                <span
                  class="comparison-label"
                >
                  SAVED
                </span>
              </div>
              <div
                class="col-1 d-flex justify-content-center align-content-center"
              >
                <div
                  class="vertical-separator m-auto"
                />
              </div>
              <div
                class="col-16"
              >
                <div
                  style="width: 100%; height: 100%;"
                >
                  <div
                    style="width: 0px; height: 0px;"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c3 row"
      >
        <div
          class="col-12"
        >
          <div
            class="c11 c12"
            id="card"
          >
            <div
              class="c13"
            >
              sector allocation
              <i
                class="c6"
              >
                <svg>
                  info.svg
                </svg>
              </i>
            </div>
            <div
              class="c14"
            >
              <div
                class="c15 row"
              >
                <div
                  class="c16 col-13"
                >
                  <canvas
                    class="chartjs-render-monitor"
                    height="0"
                    id="sector-allocation"
                    style="display: block;"
                    width="0"
                  />
                </div>
                <ul
                  class="c17 0-legend col-11"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="col-12"
        >
          <div
            class="c11 c18"
            id="card"
          >
            <div
              class="c13"
            >
              yield contribution
              <i
                class="c6"
              >
                <svg>
                  info.svg
                </svg>
              </i>
            </div>
            <div
              class="c15 row"
            >
              <div
                class="c16 col-13"
              >
                <canvas
                  class="chartjs-render-monitor"
                  height="0"
                  id="yield-contribution"
                  style="display: block;"
                  width="0"
                />
              </div>
              <ul
                class="c17 0-legend col-11"
              />
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
